examples/cloudml-sklearn-pipeline/README.md

```diff
@@ -0,0 +1,430 @@
+        |__ config.yaml             # for running normal training job on AI Platform
+        |__ hptuning_config.yaml    # for running hyperparameter tunning job on AI Platform    
+    |__ scripts
+        |__ train.sh                # convenience script for running machine learning training jobs
+        |__ deploy.sh               # convenience script for deploying trained scikit-learn model
+        |__ predict.sh              # convenience script for requesting online prediction
+        |__ predict.py              # helper function for requesting online prediction using python
+    |__ trainer                     # trainer package
+        |__ util                    # utility functions
+            |__ utils.py            # utility functions including e.g. loading data from bigquery and cloud storage
+            |__ preprocess_utils.py # utility functions for constructing preprocessing pipeline
+            |__ transform_utils.py  # utility functions for constructing transform pipeline            
+        |__ metadata.py             # dataset metadata and feature columns definitions
+        |__ constants.py            # constants used in the project
+        |__ model.py                # pre-processing and machine learning model pipeline definition
+        |__ task.py                 # training job entry point, handling the parameters passed from command line
+        |__ transform_config.py     # configuration for transform pipeline construction"        
+    |__ predictor.py                # define custom prediction behavior
+    |__ setup.py                    # specify necessary dependency for running job on AI Platform
+    |__ requirements.txt            # specify necessary dependency, helper for setup environemnt for local development
+```
+
+## Using the template
+### Step 0. Prerequisites
+Before you follow the instructions below to adapt the tempate to your machine learning job, 
+you need a Google cloud project if you don't have one. You can find detailed instructions 
+[here](https://cloud.google.com/dataproc/docs/guides/setup-project).
+
+- Make sure the following API & Services are enabled.
+    * Cloud Storage
+    * Cloud Machine Learning Engine
+    * BigQuery API
+    * Cloud Build API (for CI/CD integration)
+    * Cloud Source Repositories API (for CI/CD integration)
+
+- Configure project id and bucket id as environment variable.
+  ```bash
+  $ export PROJECT_ID=[your-google-project-id]
+  $ export BUCKET_ID=[your-google-cloud-storage-bucket-name]
+  ```
+  
+- Set up a service account for calls to GCP APIs.  
+  More information on setting up a service account can be found 
+  [here](https://cloud.google.com/docs/authentication/getting-started).
+  
+### Step 1. Tailor the scikit-learn trainer to your data
+`metadata.py` is where the dataset's metadata is defined. 
+By default, the file is configured to train on the Census dataset, which can be found at 
+`bigquery-public-data.ml_datasets.census_adult_income`.
```

Comment:
Link to BigQuery Public Datasets website?

Reply:
Added as suggested.